### ISSUE TITLE
Support a configurable suppression comment for approved one-off overrides

### ISSUE THREAD

**Issue by justinchen** [shadcn-ui/lint#19]
There is no way to exempt a single call site in code, which makes the rules hard to adopt on an existing codebase.

### Our case

Our design system allows a documented one-off override via a comment directly above `className`:

```tsx
<Button
  // design-override: translucent control over an image
  className="bg-white/80"
/>
```

Our own checker counts these in a baseline, so each new one shows up in review. When we ran `@shadcn/lint` over the same tree, **70 of its 102 `no-restyle` findings** were on call sites already approved this way. The genuinely new ones were buried.

### Ask

A configurable suppression comment, e.g.

```json
{ "settings": { "shadcn": { "suppressionComments": ["design-override", "design-token-exempt"] } } }
```

so a team's existing convention keeps working, rather than a new mechanism that requires rewriting every exemption.

### One detail worth designing around

Findings are reported per branch of a `cn()` call:

```tsx
<Button className={cn("bg-white", isActive && "bg-primary text-white", size === "lg" && "px-6")} />
```

produces several findings on different lines, so a single comment above `className` does not cover them. Whatever the mechanism, it should apply to the whole `className` expression rather than one line.

Versions: `@shadcn/lint` 0.1.0, oxlint 1.83.

**Comment by shadcn** [comment]
This is in the works.
